### DISCUSSION TITLE
Configuration options

### DISCUSSION THREAD

**Discussion by daniel-dlds** [strongswan/strongswan#1529] [General]
Hello,
I read some of the docs and the pistrong docs also, and still don't know if my scenario is supported.
I have a raspberry pi in my home, on which I want to install strongswan and pistrong to manage the certs, etc.
Then when I am away from home I want my laptop to connect to the VPN as a client, and have access to the devices on my home LAN.
I read that, one can have :
Host to host
Client server
Site to site
Is it possible to support accessing my LAN in client server mode ?
Many thanks

**Comment by tobiasbrunner** [comment]
Sure, why not? Or am I missing something?

**Reply by daniel-dlds** [reply]
Because I am new to this, just asking before I dive in. I am going to install the vpn server and work from there.

**Comment by daniel-dlds** [comment]
Got the vpn running, configured a user with pistrong. Now I want to connect from my mac laptop to the vpn.
What are the remote Id and the local ID ?
I tried some values that are configured through pistrong, but I keep getting in the syslog :
`Feb  5 00:37:42 vpnserver charon-systemd[16946]: xxx.xxx.xxx.xxx is initiating an IKE_SA
Feb  5 00:37:42 vpnserver charon-systemd[16946]: selected proposal: IKE:AES_CBC_256/HMAC_SHA2_256_128/PRF_HMAC_SHA2_256/MODP_2048
Feb  5 00:37:42 vpnserver charon-systemd[16946]: looking for peer configs matching 192.168.1.100[dlsa-faisca]...144.64.133.118[dlsa-dev]
Feb  5 00:37:42 vpnserver charon-systemd[16946]: no matching peer config found`

Where can I find the values for remote id and client id ?

**Reply by tobiasbrunner** [reply]
> What are the remote Id and the local ID ?

The IKE identities of the client and the server. These are configured in `local.id` (server/remote ID from the client's perspective) and `remote.id` (client/local ID of the client) in [swanctl.conf](https://docs.strongswan.org/docs/5.9/swanctl/swanctlConf.html). You can use wildcards if you want.

In the log message the identities are `dlsa-faisca` (server ID) and `dlsa-dev` (client ID). Make sure these match in the config and that the config is actually loaded (check with [`swanctl --list-conns`](https://docs.strongswan.org/docs/5.9/swanctl/swanctlListConns.html)). Also, if you use certificates to authenticate, the identities have to be contained as subjectAltName extension in them.

**Comment by gitbls** [comment]
Here's the answer to the same question that OP posted on https://github.com/gitbls/pistrong:

I'm not completely clear what you're asking, but here's my interpretation of it. You defined the remote IDs when you ran MakeMyCA. There are remote IDs for each of android, ios, windows, and linux. You can see them in the sudo pistrong config command, item validids. They are also visible in the config file /etc/swanctl/conf.d/pistrong-CAServerConnection.conf

The Local ID is per-user. That is, each user has a specific Local ID. You can see that by sudo pistrong list username-devname --full in the field AltNames

Once you create a Cert for your MacOS user, the file /etc/swanctl/pistrong/server-assets/username-devname.zip is created, which contains the certs for that specific client. There is also a readme.txt file in the zip file that contains the Remote ID and Local ID for that particular Cert. Also note that the Local ID for a cert is fabricated from the inputs you provided to MakeMyCA and the usernname and devname for that particular Cert.

Hope the above answers your question. If not, please respond with more specifics of what you're looking for.
